### PR TITLE
#24 docker

### LINKED ISSUE #24: Dockerize app
- [x] Node app as docker container
- [x] connect mongoDB container
- [x] connect RabbitMQ container

### LINKED ISSUE #28: Integrate RabbitMQ
Copy-paste code from @sebastiankliem's and @mmkonrad's repository.

### PR DESCRIPTION
**Pull request by sebastiankliem**
This dockerizes the node app and integrates configured containers for mongoDB and RabbitMQ. Development and testing using `docker-compose` is documented in Readme.

Could not test the AQMP/RabbitMQ integration, as there is no usable AQMP stub out there and there seems to be no way to send events from RabbitMQ in Travis.

closes #24 and closes #28

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM node:10.4.1
+
+COPY . /app
+WORKDIR /app
+RUN npm install -g nodemon
```
Comment (cmfcmf) [inline Dockerfile]: Does installing `nodemon` really make sense here? You've hard-copied the files into the Docker image, thus they'll never change.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+RUN npm install
+EXPOSE 80
```
Comment (cmfcmf) [inline Dockerfile]: Our app currently runs on port 3030.

File: README.md [hunk 3]
```diff
@@ -8,6 +8,18 @@
 
 This project uses [Feathers](http://feathersjs.com). An open source web framework for building modern real-time applications.
 
+## Development with Docker
+```
+docker-compose -f docker-compose.dev.yml up
+```
+This starts the containers for the app, mongoDB and RabbitMQ. If necessary, the app's container is build automatically in advance.
```
Comment (cmfcmf) [inline README.md]: Minor spelling issues: `MongoDB`, `built`.

File: src/channels.js [hunk 4]
```diff
@@ -51,7 +49,6 @@ module.exports = function (app) {
   // Here you can also add service specific event publishers
   // e.g. the publish the `users` service `created` event to the `admins` channel
   // app.service('users').publish('created', () => app.channel('admins'));
-
```
Comment (cmfcmf) [inline src/channels.js]: Unnecessary changes

File: src/hooks/after-achievement-module.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+          xpService.patch(uniqueCombination[0]._id, {amount: uniqueCombination[0].amount + action['amount']});
+        } else {
+          xpService.create({user_id: context.data.user_id, name: action['xp'], amount: action['amount']});
+        }
+      }
+    }
+
+    return context;
+  };
+};
```
Comment (cmfcmf) [inline src/hooks/after-achievement-module.js]: This file, for example, is outdated and duplicated code, caused by the weird history of this branch.

**Comment by sebastiankliem** [comment]
Rebasing did not work properly, so I created a new PR #44 .
